extensions/objects/models/objects.py

```diff
@@ -863,16 +869,16 @@ class NumberWithUnits(BaseObject):
             'schema': {
                 'type': 'unicode'
             }
-        }, {
-            'name': 'fraction',
-            'schema': Fraction.SCHEMA
         }, {
             'name': 'real',
             'schema': {
                 'type': 'float'
             }
         }, {
-            'name': 'unit',
+            'name': 'fraction',
+            'schema': Fraction.SCHEMA
+        }, {
+            'name': 'units',
             'schema': Units.SCHEMA
```

Comment:
This is kind of weird as to access units we have to do `ans['units']['units']`. I think we should remove Units.SCHEMA from here and keep it as a simple dict. On the other hand, NumberWithUnitsObjectFactory can be modified to store actual unitDict and not a UnitsObjectFactory object so that it is consistent with the backend.

What do you think?

Reply:
This is exactly what I have proposed initially. But after your discussion with Sean, we decided that Units should be a separate object as it may be helpful in further extension of the interaction, right? If we have to use separate Units object then, yes, we have to access it like this.

Reply:
Let's first finalize units schema because yesterday it was identfied that current schema (a dict with arbitrary key, value pairs) won't work in current framework.